### DISCUSSION TITLE
Runtime field check

### DISCUSSION THREAD

**Discussion by jmsolutionsgroup** [prisma/prisma#18989] [Q&A]
I have some fields across many models like CreatedBy, UpdatedBy, CreatedAt, UpdatedAt that I would like to update using middleware.
I've looked at the documentation for topics like [soft-delete](https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware) that utilizes code below.

```
prisma.$use(async (params, next) => {
    // Check incoming query type
    if (params.model == 'Post') {
      if (params.action == 'delete') {
        // Delete queries
        // Change action to an update
        params.action = 'update'
        params.args['data'] = { deleted: true }
      }
      if (params.action == 'deleteMany') {
        // Delete many queries
        params.action = 'updateMany'
        if (params.args.data != undefined) {
          params.args.data['deleted'] = true
        } else {
          params.args['data'] = { deleted: true }
        }
      }
    }
    return next(params)
  })
```

The problem that I have is that I want to make it generic and check the params.model for 50+  tables.  My other issue is that not all of my tables have these properties  I want to be able to check the model and if it has the createdBy field, then set it.   I've looked into a couple different ways of completing my task.   

I've looked at the [client-extenstions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) as a way to extend and then utilize the [prisma validator](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/prisma-validator).  I've also looked at importing @prisma/internals and utilizing `getDMMF` to analyze the schema and create the type.  

I have been unsuccessful in both directions.   I would like to be able to create a piece of middleware similar to below.

```
  prisma.$use(async (params, next) => {
    const { model, action, args } = params;

    switch (action) {
      case "create":
        const obj: typeof model = model as typeof model; //cast or get the object type
        if (!!obj && obj.hasOwnProperty("createdById")) { //check to see if the object has the property
          params.args.data["createdById"] = userId; //set the property value
        }

        break;
    }

    return next(params);
  });
```
Any help is appreciated

**Comment by nurul3101** [comment]
Hello @uchetfield 👋 

I don't think there is a built-in way to introspect the schema or the properties of a model during the runtime. I think you would need to create a manual mapping between your models and their properties to do what you're trying to do. I'll investigate and check if there's any way through which you could achieve this.

**Reply by nurul3101** [reply]
@uchetfield Did you had a chance to look at my last response?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@uchetfield We are closing this Discussion due to inactivity. Please open a new Discussion if you have the a similar problem :pray:
